plugins/modules/win_domain_computer.py

```diff
@@ -86,6 +89,28 @@
     type: str
     choices: [ absent, present ]
     default: present
+  offline_domain_join:
+    description:
+      - Provisions a computer in the directory and provides a BLOB file that can be used on the target computer/image to join it to the domain while offline.
+      - If the computer already exists, no BLOB will be created/returned, and the module will operate as it would have without offline domain join.
+    type: bool
+  odj_return_blob:
+    description:
+      - Returns the BLOB in the module output. Note that the BLOB should be treated as a secret, as it contains the machine password.
+      - The module should be called with C(no_log) when using this option.
+      - The BLOB file contains a terminating null character which will not be included. See Notes.
+  odj_blob_path:
+    description:
+      - The path to the file where the BLOB will be saved.
+      - Required when I(offline_domain_join=True) and I(odj_return_blob=False).
+      - If using I(odj_return_blob=True) this can be omitted, and a temporary file will be used and then destroyed.
+      - If I(odj_return_blob=True) and this is specified, it will be used and file will remain.
+notes:
+  - "For more information on Offline Domain Join
+    see U(https://docs.microsoft.com/en-us/previous-versions/windows/it-pro/windows-server-2008-R2-and-2008/dd392267(v=ws.10))"
+  - When using the ODJ BLOB to join a computer to the domain, it must be written out to a file.
```

Comment:
Is this a limitation of `djoin.exe`? Is it a problem when you PInvoke?

Reply:
Do you mean having it written to a file? That's a limitation of `djoin.exe`. With p/invoke you'd be giving it a buffer directly (probably the base64-decoded contents of what we get from `djoin.exe`); but that would happen outside this module anyway.

Reply:
Yep, was just curious if having it written to a file was a limitation of what djoin.exe offers or whether the underlying Win32 calls also has the restrictions.

Reply:
Oh I see on output? That's also a limitation of `djoin.exe` (sort of). It has an option to write the content to stdout, but you still have to give it a file to write to anyway, and the stdout has other stuff with it, so it's not that useful for programmatic use. The API doesn't deal with files at all, so on the provisioning part the API would have given the data back in a buffer too.

Reply:
No worries, I was curious because I've never used this method before.